File: cine/JiaChengZhang_Cine/JCZhang-Cine-JAVA/github/Cine_JAVA/src/main/java/models/Butaca.java
package models;

import enums.ESTADOS;

public class Butaca {
    public ESTADOS estado;

    Butaca(ESTADOS estado){
        this.estado = estado;
    }
    static Butaca FREE_SEAT = new Butaca(ESTADOS.LIBRE);
    static Butaca RESERVED_SEAT = new Butaca(ESTADOS.RESERVADO);
    static Butaca SOLD_SEAT = new Butaca(ESTADOS.OCUPADO);
    static  Butaca VIP_SEAT = new Butaca(ESTADOS.VIP);

    @Override
    public String toString() {
        return switch (estado){
            case LIBRE -> "💺";
            case OCUPADO -> "🍿";
            case RESERVADO -> "❌";
            case VIP -> "🌟";
        };
    }
}

File: cine/JiaChengZhang_Cine/JCZhang-Cine-KOTLIN/Cine_Kotlin/src/main/kotlin/models/Butaca.kt
package models

import enums.ESTADOS

// El estado base de las butacas es libre.
data class Butaca(var estado: ESTADOS = ESTADOS.LIBRE){

    override fun toString(): String {
        return when(Butaca(estado)){
            Butaca(estado = ESTADOS.LIBRE) -> "💺"
            Butaca(estado = ESTADOS.RESERVADO) -> "❌"
            Butaca(estado = ESTADOS.OCUPADO) -> "🍿"
            else -> ({ println("Fatal Error!!!!")}).toString()
        }
    }
}


